src/components/DataplanePolicies/DataplanePolicies.vue

```diff
@@ -175,6 +172,30 @@ export default {
           dppName: this.dppName,
         })
 
+        for (const item of items) {
```

Comment:
ok, so this maps policies from the API to our internal representation, right? I'd have it in a separate function or maybe not mutate the data coming from the API but return a new object?

Reply:
Yes, correct. When displaying this content before, we used the original `items` data structure and the methods `getPolicyTitle` and `getPolicyLink` to extract additional data. Calling methods from templates is one of the most common pitfalls the cause additional work: when a component’s template is being re-rendered, all method calls _must_ be executed again because there is no way to tell whether its result changed. By computing this data and adding it to the `items` data structure, this is avoided.

Now as for re-using the `items` data structure as it is returned by the API call, I generally agree with the approach to mutate data. In this case, I don’t see a problem however because `items` is _completely_ local to this function. `items` _is_ completely new and there can also not be any other references to it. So mutating it directly is more direct than creating a new structure. One could make an argument about it being less resilient for certain types of refactorings, but that’s not that pressing of a matter here.

In any case, externalizing this into a separate method is definitely making this a bit cleaner. I’ve done that.